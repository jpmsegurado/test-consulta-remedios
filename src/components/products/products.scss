.products {
  width: 100%;
  display: flex;
  flex-wrap: wrap;

  > .product-wrapper {
    width: calc(100% / 3);
    padding: 15px;
    box-sizing: border-box;
  }
}