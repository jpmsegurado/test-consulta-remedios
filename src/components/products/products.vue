<template>
  <div class="products">
    <div class="product-wrapper" v-for="product in products" :key="product.id">
      <product :product-info="product" />
    </div>
  </div>
</template>

<script>
import product from "../product/product";
import { mapGetters, mapActions } from "vuex";

export default {
  components: {
    product
  },
  computed: {
    ...mapGetters({
      products: "PRODUCT/ALL"
    })
  },
  methods: {
    ...mapActions({
      loadProducts: "PRODUCT/LOAD_PRODUCTS"
    })
  },
  mounted() {
    this.loadProducts();
  }
};
</script>

<style lang="scss" scoped src="./products.scss" />
