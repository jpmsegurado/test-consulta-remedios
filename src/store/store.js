import Vue from "vue";
import Vuex from "vuex";
import PRODUCT from "./modules/products";
import CART from "./modules/cart";

Vue.use(Vuex);

export default new Vuex.Store({
  modules: {
    PRODUCT,
    CART
  }
});
