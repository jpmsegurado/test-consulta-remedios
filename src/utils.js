export function formatMoney(value, decimal = 2) {
  const formatter = new Intl.NumberFormat("pt-BR", {
    style: "currency",
    currency: "BRL",
    minimumFractionDigits: decimal
  });

  return formatter.format(value);
}
