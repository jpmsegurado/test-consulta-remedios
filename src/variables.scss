$product-bg: #EEEEEE;
$product-name: #746A6A;
$product-price: #3486E6;
$cart-border: #E1E1E1;
$cart-text: #746A6A;
$item-cart-name: #7F7575;
$white: #fff;
$cart-button: #54A3FF;
$select-border: #E1E1E1;

.fade-enter-active, .fade-leave-active {
  transition: opacity .3s;
}
.fade-enter, .fade-leave-to {
  opacity: 0;
}