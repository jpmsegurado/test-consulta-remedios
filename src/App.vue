<template>
  <div id="app">
    <div class="row">
      <div class="products-wrapper">
        <h2>Games</h2>
        <mySelect v-model="selected" :value="selected" :list="options" />
        <products :selectedOption="options[selected]" />
      </div>

      <div class="shopping-cart-wrapper">
        <shopping-cart />
      </div>
    </div>
  </div>
</template>

<script>
import products from "./components/products/products";
import shoppingCart from "./components/shopping-cart/shopping-cart";
import mySelect from "./components/select/select";

export default {
  name: "app",
  components: {
    products,
    shoppingCart,
    mySelect
  },
  data() {
    return {
      options: [
        { label: "Mais populares", prop: "score" },
        { label: "Mais baratos", prop: "price" },
        { label: "Ordem Alfab√©tica", prop: "name" }
      ],
      selected: 0
    };
  }
};
</script>

<style lang="scss">
@import url("https://fonts.googleapis.com/css?family=Open+Sans");
body {
  padding: 0;
  margin: 0;
}
#app {
  font-family: "Open Sans";
  padding: 42px 70px;

  > .row {
    display: flex;
    width: 100%;
    flex-wrap: wrap;

    .products-wrapper {
      width: 75%;

      > h2 {
        font-size: 48px;
        margin-top: 0;
        margin-bottom: 44px;
        width: calc(100% - 277px);
        display: inline-block;
      }

      > .select {
        width: 262px;
        margin-right: 15px;
      }
    }

    .shopping-cart-wrapper {
      width: 25%;
    }
  }
}
</style>
