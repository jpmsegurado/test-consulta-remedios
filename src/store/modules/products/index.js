import products from "../../../assets/products.json";

const state = {
  products: []
};

const getters = {
  ALL: state => state.products
};

const actions = {
  LOAD_PRODUCTS: ({ commit }) => {
    return Promise.resolve(commit("SET_PRODUCTS", products));
  }
};

const mutations = {
  SET_PRODUCTS: (state, products) => {
    state.products = products;
  }
};

export default {
  namespaced: true,
  state,
  getters,
  actions,
  mutations
};
