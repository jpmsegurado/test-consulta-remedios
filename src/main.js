import Vue from "vue";
import App from "./App.vue";
import store from "./store/store";

Vue.config.productionTip = false;

Vue.prototype.formatMoney = (value, decimal = 2) => {
  const formatter = new Intl.NumberFormat("pt-BR", {
    style: "currency",
    currency: "BRL",
    minimumFractionDigits: decimal
  });

  return formatter.format(value);
};

new Vue({
  store,
  render: h => h(App)
}).$mount("#app");
