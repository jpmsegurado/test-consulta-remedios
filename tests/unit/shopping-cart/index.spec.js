import { shallowMount } from "@vue/test-utils";
import shoppingCart from "@/components/shopping-cart/shopping-cart.vue";
import products from "@/assets/products.json";
import { formatMoney } from "../../../src/utils";
import store from "@/store/store";

describe("shopping-cart-item.vue", () => {
  it("render shopping cart empty state", () => {
    const wrapper = shallowMount(shoppingCart, {
      store,
      mocks: {
        formatMoney
      }
    });

    expect(wrapper).toMatchSnapshot();
  });

  it("check shippingCost with one product", () => {
    const wrapper = shallowMount(shoppingCart, {
      store,
      mocks: {
        formatMoney
      }
    });

    store.dispatch("CART/ADD_TO_CART", products[0]);
    expect(wrapper.vm.shoppingCost).toBe(10);
  });

  it("check shippingCost with three product", () => {
    const wrapper = shallowMount(shoppingCart, {
      store,
      mocks: {
        formatMoney
      }
    });

    store.dispatch("CART/ADD_TO_CART", products[0]);
    store.dispatch("CART/ADD_TO_CART", products[0]);
    store.dispatch("CART/ADD_TO_CART", products[1]);

    expect(wrapper.vm.shoppingCost).toBe(0);
  });
});
