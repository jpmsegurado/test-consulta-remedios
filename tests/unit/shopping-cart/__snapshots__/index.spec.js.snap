// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shopping-cart-item.vue render shopping cart empty state 1`] = `
<div class="shopping-cart" style="top: 0px;">
  <h3>
    Carrinho
    <!---->
  </h3>
  <div class="cart-icon"><img src="@/assets/cart-icon.svg" alt="Shopping Cart"></div>
  <p>
    Até o momento, <br>
    o seu carrinho está vazio
  </p>
  <!---->
</div>
`;
